ksql-engine/src/test/java/io/confluent/ksql/schema/inference/DefaultSchemaInjectorTest.java

```diff
@@ -416,6 +416,37 @@ public void shouldThrowOnUnsupportedType() {
     }
   }
 
+  @Test
+  public void shouldEscapeAvroSchemaThatHasReservedColumnName() {
```

Comment:
FYI, escaping a reserved word with quotes doesn't actually work unless the field name is upper case.

I'm not saying that negates what you're doing. Just something to keep in mind.  I _may_ be better to throw an error to the user if a non-upper-cased reserved word is detected, as this would give a better user-experience that shit just not working.

'course, you're tackling the quoted stuff next anyway.

Reply:
The only reason I'm doing this is because we have some unit tests around this that would start failing - so it's either change the code to maintain backwards compatibility or change the tests.